jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ServiceAccountEnricher.java

```diff
@@ -29,55 +32,77 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 public class ServiceAccountEnricher extends BaseEnricher {
+    @AllArgsConstructor
+    public enum Config implements Configs.Config {
+        SKIP_CREATE("skipCreate", "false");
+
+        @Getter
+        protected String key;
+        @Getter
+        protected String defaultValue;
+    }
+
     public ServiceAccountEnricher(JKubeEnricherContext enricherContext) {
         super(enricherContext, "jkube-serviceaccount");
     }
 
     @Override
     public void create(PlatformMode mode, KubernetesListBuilder builder) {
         Map<String, String> deploymentToSaPair = new HashMap<>();
-        List<ServiceAccount> serviceAccounts = new ArrayList<>();
 
-        // Check XML config and see if there are any service accounts specified
-        ResourceConfig xmlResourceConfig = getConfiguration().getResource();
-        if(xmlResourceConfig != null && xmlResourceConfig.getServiceAccounts() != null) {
-            for(ServiceAccountConfig serviceAccountConfig : xmlResourceConfig.getServiceAccounts()) {
-                if(serviceAccountConfig.getName() != null) {
-                    serviceAccounts.add(createServiceAccount(builder, serviceAccountConfig.getName()));
-                }
-                if(serviceAccountConfig.getDeploymentRef() != null) {
-                    deploymentToSaPair.put(serviceAccountConfig.getDeploymentRef(), serviceAccountConfig.getName());
+        // Check config and see if there are any service accounts specified
+        ResourceConfig resourceConfig = getConfiguration().getResource();
+        if(resourceConfig != null && resourceConfig.getServiceAccounts() != null) {
+            deploymentToSaPair.putAll(resourceConfig.getServiceAccounts()
+                .stream()
+                .filter(sa -> sa.getDeploymentRef() != null)
+                .collect(Collectors.toMap(ServiceAccountConfig::getDeploymentRef, ServiceAccountConfig::getName)));
+        }
+        builder.addAllToServiceAccountItems(createServiceAccountFromResourceConfig(resourceConfig));
+        builder.addAllToServiceAccountItems(createServiceAccountsReferencedInDeployment(builder, deploymentToSaPair));
+    }
+
+    private List<ServiceAccount> createServiceAccountFromResourceConfig(ResourceConfig resourceConfig) {
+        List<ServiceAccount> serviceAccounts = new ArrayList<>();
+        if(resourceConfig != null && resourceConfig.getServiceAccounts() != null) {
+            for(ServiceAccountConfig serviceAccountConfig : resourceConfig.getServiceAccounts()) {
+                if(serviceAccountConfig.getName() != null
+                    && !Boolean.parseBoolean(getConfig(Config.SKIP_CREATE))) {
```

Comment:
Checking for the skip create flag so late seems like a waste of resources. I'd either move this check to the calling method:
```java
if (!Boolean.parseBoolean(getConfig(Config.SKIP_CREATE)) {
 
 builder.addAllToServiceAccountItems(createServiceAccountFromResourceConfig(resourceConfig));
}
```

or to line 70 if clause